### PR TITLE
GEODE-6032: Updated javadoc to clarify what is communicated with hasD…

### PR DESCRIPTION
**Pull request by agingade**
### For all changes:
- [Yes] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [Yes] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [Yes] Is your initial contribution a single, squashed commit?

- [Yes] Does `gradlew build` run cleanly?

- [NA] Have you written or updated unit tests to verify your changes?

- [NA] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/Delta.java [hunk 1]
```diff
@@ -32,7 +32,9 @@
 public interface Delta {
 
   /**
-   * Returns true if this object has pending changes it can write out.
+   * Returns true if this object has delta to be applied.
+   * If true, delta is applied and delta data is sent between client/server, otherwise entire
+   * data is sent between client/server.
```
Comment (bschuchardt) [inline geode-core/src/main/java/org/apache/geode/Delta.java]: I'm sorry but this javadoc doesn't seem at all clear to me and it has grammatical errors.  I think you should keep the original Javadoc and append something like this:
_Returns false if pending changes are not available, which will cause the entire object to be transmitted to other processes._

Comment (agingade) [reply geode-core/src/main/java/org/apache/geode/Delta.java]: Thanks Bruce. I wanted to make it clear what is communicated based on hasDelta().

Comment (kirklund) [reply geode-core/src/main/java/org/apache/geode/Delta.java]: Here's my version:
```
Returns true if this object has pending changes that can be transmitted as a delta.
```
If that's not enough then maybe:
```
Returns true if this object has pending changes that can be transmitted as a delta. Returns false if this object must be transmitted in its entirety.
```

File: geode-core/src/main/java/org/apache/geode/Delta.java [hunk 2]
```diff
@@ -32,9 +32,8 @@
 public interface Delta {
 
   /**
-   * Returns true if this object has delta to be applied.
-   * If true, delta is applied and delta data is sent between client/server, otherwise entire
-   * data is sent between client/server.
+   * Returns true if this object has pending changes it can write out as delta.
```
[the remark this answers is not in the document]
Comment (agingade) [reply geode-core/src/main/java/org/apache/geode/Delta.java]: New commit has the requested changes.

**Pull request merged**